### ISSUE TITLE
Ignore UndocumentedPublicFunction on Composable Preview

### ISSUE THREAD

**Issue by igorescodro** [detekt/detekt#3093]
In my opinion, the functions annotated with both `@Composable` and `@Preview` should not require public documentation.

## Expected Behavior of the rule

For example the following code should not trigger `UndocumentedPublicFunction`:
```
@Preview(showBackground = true)
@Composable
fun AlkaaTopBarPreview() {
    AlkaaTheme {
        AlkaaTopBar(HomeSection.Tasks)
    }
}
```

If the functions are set to `private` it will trigger `UnusedPrivateMember`.
I think it does not aggregate for an public API documenting it.

Thanks for your time.

**Comment by schalkms** [comment]
Annotation excluding needs to be a cross cutting concern as discussed and proposed in #2231. Please vote for this feature.

We decided to not include framework specifics in configuration options as for instance the `Dagger` framework cases. This adds to the complexity of the rules and creates more maintenance effort. Furthermore, users not using `Dagger` do not profit from such config options.
Such framework specifics wrapped in rules are always based on opinions, which should be avoided as a result of the decision.

For now, you could copy the rule and implement this condition in a [custom rule](https://detekt.github.io/detekt/extensions.html#custom-rulesets), if you need it desperately.
Moreover, you could also use the `@Ignore` annotation.

Due to the mentioned arguments, I'm closing this.

**Comment by schalkms** [comment]
Thanks for bringing some insights on how detekt is being used.

Annotation excluding needs to be a cross cutting concern as discussed and proposed in #2231. Please vote for this feature.

We decided to not include framework specifics in configuration options as for instance the `Dagger` framework cases. This adds to the complexity of the rules and creates more maintenance effort. Furthermore, users not using `Dagger` do not profit from such config options.
Such framework specifics wrapped in rules are always based on opinions, which should be avoided as a result of the decision.

For now, you could copy the rule and implement this condition in a [custom rule](https://detekt.github.io/detekt/extensions.html#custom-rulesets), if you need it desperately.
Moreover, you could also use the `@Ignore` annotation.

Due to the mentioned arguments, I'm closing this.

**Comment by ezamelczyk** [comment]
What @Ignore annotation? It doesn't exist.

**Comment by BraisGabin** [comment]
I assume that he meant `@Suppress`. And the last version of detekt allows you to add `ignoreAnnotated` to the configuration of any rule so you can configure detekt as you prefer.

We also have this page in our documentation: https://detekt.github.io/detekt/compose.html

**Comment by schalkms** [comment]
Yes, I meant the **Suppress** annotation. I updated my previous post.
